drivers/volume/portworx/portworx.go

```diff
@@ -1992,6 +2006,35 @@ func (d *portworx) IsStorageExpansionEnabled() (bool, error) {
 
 // IsPureVolume return true if volume is pure volume else false
 func (d *portworx) IsPureVolume(volume *torpedovolume.Volume) (bool, error) {
+	var proxySpec *api.ProxySpec
+	var err error
+	if proxySpec, err = d.getProxySpecForAVolume(volume); err != nil {
+		return false, err
+	}
+	if proxySpec == nil {
+		return false, nil
+	}
+
+	if proxySpec == nil {
+		return false, nil
+	}
+	if proxySpec.ProxyProtocol == api.ProxyProtocol_PROXY_PROTOCOL_PURE_BLOCK {
+		logrus.Debugf("Volume is Pure Block volume: %s", volume.ID)
+		return true, nil
+	}
+
+	if proxySpec.ProxyProtocol == api.ProxyProtocol_PROXY_PROTOCOL_PURE_FILE {
```

Comment:
If condition in line 2021 is true, this line will never be executed as the function would return from line 2021.

If we need to check only one of the conditions, a simpler/better way would be:

```
if proxySpec.ProxyProtocol == api.ProxyProtocol_PROXY_PROTOCOL_PURE_BLOCK  || proxySpec.ProxyProtocol == api.ProxyProtocol_PROXY_PROTOCOL_PURE_FILE {
```